pulsar-client-admin-api/src/main/java/org/apache/pulsar/client/admin/Topics.java

```diff
@@ -2370,23 +2370,27 @@ void setInactiveTopicPolicies(String topic
      * Set message-dispatch-rate (topic can dispatch this many messages per second).
      *
      * @param topic
+     * @param resetMode
+     *            clear the previous rate and reset (default true). if it is false, merge the two rate
      * @param dispatchRate
      *            number of messages per second
      * @throws PulsarAdminException
      *             Unexpected error
      */
-    void setDispatchRate(String topic, DispatchRate dispatchRate) throws PulsarAdminException;
```

Comment:
Here and below we are introducing breaking API changes
You can deprecate this method and add a new one

Reply:
The method `setDispatchRate(String topic, DispatchRate dispatchRate)` will not be used anywhere after introducing new option `--reset-mode`, and its implementation will also change in order to adapt to the new option, so I think it could be possible to directly replace it with `setDispatchRate(String topic, boolean resetMode, DispatchRate dispatchRate)`. Maybe we can ask other people's ideas?

Reply:
Pulsar Admin API is a public API that users can use from Java code.

It is not only used by the pulsar-admim tool